browser/ai_chat/ai_chat_api.cc

```diff
@@ -0,0 +1,120 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ai_chat/ai_chat_api.h"
+
+#include <base/containers/flat_map.h>
+
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "brave/browser/ai_chat/buildflags.h"
+#include "brave/browser/ai_chat/constants.h"
+#include "brave/components/constants/brave_services_key.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "url/gurl.h"
+
+namespace {
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
```

Comment:
since this feature is available in private windows, can someone ensure:

1. no requests are cached client-side in a way that might leak to disk
2. all network requests are going through Tor if done in a Tor window

Reply:
> 1. no requests are cached client-side in a way that might leak to disk

`api_request_helper` will use these `load_flags`: `net::LOAD_BYPASS_CACHE | net::LOAD_DISABLE_CACHE | net::LOAD_DO_NOT_SAVE_COOKIES` so I think we're ok there